dicom_validator/tests/validator/test_iod_validator.py

```diff
@@ -29,10 +33,20 @@ def new_data_set(tags, ds: Optional[Dataset] = None):
                 items.append(new_data_set(item_tags, data_set))
             value = Sequence(items)
         data_set[tag] = DataElement(tag, vr, value)
-    data_set.file_meta = FileMetaDataset()
-    data_set.is_implicit_VR = False
+    # write the dataset into a file and read it back to ensure the real behavior
+    if "SOPInstanceUID" not in data_set:
+        data_set.SOPInstanceUID = "1.2.3"
     data_set.is_little_endian = True
-    return data_set
+    data_set.is_implicit_VR = True
+    fp = DicomBytesIO()
+    kwargs = (
+        {"write_like_original": False}
+        if int(pydicom.__version_info__[0]) < 3
+        else {"enforce_file_format": True}
+    )
```

Comment:
pydicom 3 does not support Python < 3.10, so I had to add support for pydicom 2.x